pycloudlib/instance.py

```diff
@@ -28,12 +28,17 @@ def __init__(self, key_pair):
         self._sftp_client = None
         self._tmp_count = 0
 
-        self.name = ''
         self.boot_timeout = 120
         self.key_pair = key_pair
         self.port = '22'
         self.username = 'ubuntu'
 
+    @property
+    def name(self):
+        """Return instance name."""
+        self.name = ''
+        return self.name
```

Comment:
We can't set self.name within the @property method name as there is no setter. You'd end up getting the following error on BaseInstance:

```Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 6, in name
AttributeError: can't set attribute
```

If we just return an empty string we should be good here. Let subclasses implement this at will

```suggestion
        return ''
```